### PR TITLE
AGENT-99: docker syslog/json smoketests

### PR DESCRIPTION
**Pull request by echee2**
This PR adds smoketests for docker json & docker syslog.

CircleCi launches 3 images -- agent, uploader, verifier.  All processes synchronize on scalyr backend state in qatesting.

Note: The python3 `smoketest.py` code is not included in this PR, pending further stabilization and inclusion of k8s testing capability.  In the near future, we will move `smoketest.py` into this repo.  (This PR is really about getting CircleCI to launch the currently self-contained docker image that in turn contains `smoketest.py`.)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -44,49 +43,93 @@ jobs:
...
-  smoke27:
+  smoke-standalone-27:
     docker:
       - image: circleci/python:2.7-jessie-browsers
     steps:
       - setup_remote_docker:
           docker_layer_caching: true
       - run:
-          command: docker run -it -e TEST_BRANCH=${CIRCLE_BRANCH} -e SCALYR_MAX_QUERY_SEC=300 -e PYTHON_VERSION=2.7 -e SCALYR_API_KEY=${SCALYR_API_KEY} -e READ_API_KEY=${READ_API_KEY} -e SCALYR_SERVER=${SCALYR_SERVER} scalyr/scalyr-agent-ci-smoketest /tmp/smoketest.sh
+          command: docker run -it -e TEST_BRANCH=${CIRCLE_BRANCH} -e MAX_WAIT=300 -e PYTHON_VERSION=2.7 -e SCALYR_API_KEY=${SCALYR_API_KEY} -e READ_API_KEY=${READ_API_KEY} -e SCALYR_SERVER=${SCALYR_SERVER} scalyr/scalyr-agent-ci-smoketest:v1.1 /tmp/smoketest_standalone.sh
```
Comment (imron) [inline .circleci/config.yml]: (discussion only) what versioning scheme are you using here?

File: .circleci/smoketest_docker.sh [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+# container names for all test containers
+# The suffixes MUST be one of (agent, uploader, verifier) to match verify_upload::DOCKER_CONTNAME_SUFFIXES
+contname_agent="ci_agent_docker_${syslog_or_json}_agent"
+contname_uploader="ci_agent_docker_${syslog_or_json}_uploader"
+contname_verifier="ci_agent_docker_${syslog_or_json}_verifier"
+
+
+# Kill leftover containers and cleanup workdir
+/bin/rm -rf $workdir
```
[the remark this answers is not in the document]
Comment (echee2) [reply .circleci/smoketest_docker.sh]: Good idea to protect against unintentional deletion. Will do.

File: .circleci/config.yml (hunk 1 above)
Comment (echee2) [reply .circleci/config.yml]: To be honest I didn't have time to put any thought into choosing a scheme, only that it should be something static, not `latest`, and something we can increment with different revisions.  Any preference/suggestions?

File: .circleci/smoketest_docker.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (echee2) [reply .circleci/smoketest_docker.sh]: I can't verify the changes now that qatesting is rejecting requests.  However, I've made a change where workdir and branch are no longer passed to the script.  (Hence no need to delete workdir.) 

The main idea is that CircleCI checks out a copy of the code and we make a change to the VERSION file in-situ.  So for development, a driver script should really do the same -- checkout the branch (or rsync from dev branch) into a local tmp directory.

**Pull request merged**